subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveCachingStateStep.java

```diff
@@ -69,6 +71,8 @@ public CachingResult execute(UnitOfWork work, BeforeExecutionContext context) {
         CachingState cachingState;
         if (!buildCache.isEnabled() && !buildScansEnabled) {
             cachingState = BUILD_CACHE_DISABLED_STATE;
+        } else if (context.getValidationProblems().isPresent()) {
+            cachingState = VALIDATION_FAILED_STATE;
```

Comment:
I don't think that'd be for the better. We already logged the validation problems as deprecation warnings. There might be multiple validation problems, too, and merging those is not nice.

Reply:
I am thinking how the user is going to find out what validation failed when caching (or incremental build) has been disabled. The message "Validation failed" doesn't give me any hint. If I don't use a build scan and run without `--info`, then I won't get any information at all that there has been a validation problem. If I use build scans, then I'll see the `Validation failed` message, though I won't know that I need to find the message on the deprecations page, nor how to find the task in question on the deprecations page.

This task emits a deprecation warning: https://ge.gradle.org/s/hn56auiathetk/timeline?details=75dpsv4ioedm2
You can't see from the documentation page which deprecation it is: https://ge.gradle.org/s/hn56auiathetk/deprecations

It seems to me very cheap to add the failed validations to the caching disabled reason and the out-of-date message and it would make it much better for the user.

I am not sure what we should do for users which don't use build scans. Printing another summary at the end seems redundant, since we already have the deprecation warnings summary. Maybe we could print a different summary text when the deprecations affected performance?